### PR TITLE
Add python script to generate ISBN range table in isbn.js

### PR DESCRIPTION
**Pull request by aurimasv**
See https://www.isbn-international.org/range_file_generation
Re zotero/zotero#626

### CODE REVIEW

File: generate-isbn-ranges.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+import urllib.request
+import xml.etree.ElementTree as ET
+import sys
+
+# See https://www.isbn-international.org/range_file_generation
+url = 'http://www.isbn-international.org/export_rangemessage.xml'
```
Comment (dstillman) [inline generate-isbn-ranges.py]: Should be 'https'.

Comment (aurimasv) [inline generate-isbn-ranges.py]: I think they have issues with their certificate. Though IIRC python doesn't care

Comment (aurimasv) [inline generate-isbn-ranges.py]: Or they did when I was looking at this before. I remember considering HTTP vs HTTPS

Comment (dstillman) [inline generate-isbn-ranges.py]: Well, they're currently serving a 301 redirect from HTTP to HTTPS.

**Comment by dstillman** [comment]
Is there any data that can be returned from the response that can break isbn.js? If this is going to be in the build script, either the Python has to fail (which will abort the build) or the JS has to fail gracefully in such a way that ISBN hyphenation stops working but otherwise doesn't break Zotero startup.

**Comment by aurimasv** [comment]
Hmmm, I can make this better if we want to automate the update (I think that's a good idea). Would you prefer that if the update fails, the old isbn.js is kept and an error is reported to the console, or would you rather abort the build?

**Comment by dstillman** [comment]
Yeah, I'd say we only overwrite isbn.js if generate-isbn-ranges.py has a return code of 0 and js -C on the file passes. (That's SpiderMonkey's compile-only mode.)

**Comment by aurimasv** [comment]
I tweaked the script to take a command line argument, which should point to the output file (otherwise outputs to stdout). The file is only written if the python script does not throw. I also added some sanity checking to make sure that we're only dealing with integers, which means that no response data can break the syntax in the isbn.js file.

Lastly, I de-compacted the JSON output to make diffs more comprehensible. It seems that these ranges are updated more often than expected.

**Comment by dstillman** [comment]
Can you set the executable bit on the file?

**Comment by aurimasv** [comment]
Done. I always forget...

**Comment by dstillman** [comment]
Thanks. Can squash.

**Comment by aurimasv** [comment]
done

**Comment by dstillman** [comment]
Thanks!

**Pull request merged**